roles/haproxy/templates/etc/haproxy/haproxy_openstack.cfg

```diff
@@ -149,6 +149,9 @@ frontend {{ name }}
   bind :::{{ enc_port }} ssl crt /etc/haproxy/openstack.pem no-sslv3 ciphers AES128-SHA:AES256-SHA
   default_backend {{ name }}
   reqadd X-Forwarded-Proto:\ https
+  {% if name=="keystone" -%}
+  reqadd X-Forwarded-Port:\ 5000
```

Comment:
This is needed because of the way we are doing multiple OP providers. The configuration for openid-connect is stored in files and because of this the client is redirected to port 5002 instead of the external port 5000, after the OIDC handshake done is.